### ISSUE TITLE
Show previews from subfolders if folder doesn't contains any images

### ISSUE THREAD

**Issue by BloodyIron** [nextcloud/photos#489]
When using the folder view, the folders that doesn't contain images are showed blank with the folder icon. They should show the folder's subfolder's images if possible

<details>
<summary>Old description (Click to expand)</summary>
Can't remember which version the change happened, but I really miss the old per-folder gallery method, instead of everything strewn together in one big list of pictures. It's really disorganised and painful to use now. I really want to see the old structure style as an option (or by default), not just how it is now. It was really annoying the change was done without considering how well the previous method worked.

Can we please get this as an option?
</details>

**Comment by skjnldsv** [comment]
Hey, you have it already as "albums" on the navigation.

**Comment by BloodyIron** [comment]
That's not the same, previously you would get images as previews of what's in the folder, and now it's just a folder icon. Please reopen as that is definitely not the same (but almost).

**Comment by skjnldsv** [comment]
> That's not the same, previously you would get images as previews of what's in the folder, and now it's just a folder icon.

You do get images as preview on folders.
It does it if you have images directly within the folder :)

**Comment by skjnldsv** [comment]
![Capture d’écran_2020-10-17_17-38-19](https://user-images.githubusercontent.com/14975046/96347084-8f99d500-109f-11eb-8861-bf3d7d90335e.png)

**Comment by BloodyIron** [comment]
That's most definitely not the same behaviour. Previously it was recursive, not just one folder deep. I have many folders inside folders inside folders, etc. I really do need recursive previewing, just like it was before the major Photos gallery change, currently this is not what works for me.

**Comment by skjnldsv** [comment]
You're right, this is slightly different.
Let me adapt this ticket :)

**Comment by BloodyIron** [comment]
That might work for me. I run a whole lot of events and also contain personal pictures in many different folders/subfolders, etc, with a complex (but organised) folder tree structure. So having this behaviour really helps me rapidly drill down to where I want to go. Also, not sure if this is relevant, but in the past I relied on the periodic preview generation cron job to make it so the previews were generally available for me by the time I went to use them.

I do appreciate you trying to work with me on clarifying what I'm seeking here. :) I don't exactly want to roll all the hard work back, but the previous behaviour really worked so well for me, and from what I've heard, others too.
